src/lib.rs

```diff
@@ -28,6 +30,8 @@ use cast::{u32, u64};
 use i2cdev::core::I2CDevice;
 use spidev::SpidevTransfer;
 
+pub mod serial_impl;
```

Comment:
I don't think we should expose this module. Re-exporting the `Serial` wrapper would probably be better.

Reply:
Fixed with your commits.